kernel/nsh/builtins/info.c

```diff
@@ -62,13 +66,39 @@ static void info_idt() {
 	kprintf("Size = %u bytes\n", sizeof(idt));	
 }
 
-static void info_buddy() {
+static void info_buddy_print_order(int order) {
+	uint8_t oldcolor = sb_get_color(sb_current);
+	kprintf("order %u(size of %u)\n", order, buddy->orders[order].size);
+	for (size_t i = 0; i < buddy->orders[order].size; i++) {
+		if (i % 32 == 0) {
+			kprintf("\n%4x  ", i);
+		}
+		bitmap_t val = buddy->orders[order].bitmap[i];
+		if (val == 0) {
+			sb_set_bg(sb_current, SB_COLOR_WHITE);
+			kprintf("  ");
+		} else if (val == 0xff) {
+			sb_set_bg(sb_current, SB_COLOR_RED);
+			kprintf("  ");
+		} else {
+			sb_set_bg(sb_current, SB_COLOR_BROWN);
+			kprintf("%2x", val);
+		}
+		sb_set_color(sb_current, oldcolor);
+	}
+	sb_set_color(sb_current, oldcolor);
+	kprintf("\n");
+}
+
+static void info_buddy(int order) {
 	kprintf("INFO BUDDY\n");
 	kprintf("buddy address:        %p\n", buddy);
 	kprintf("buddy size:           %u KB\n", buddy->size / 1024);
```

Comment:
I think we could check the value of ```buddy->size / 1024``` and if it equals 0 we should keep the size value in bytes or else keep it in kilobytes. This way the buddy print will be modular according to the system memory.

Reply:
Agree

Reply:
not sure it's really important to know the exact size of the buddy, what really matter is the size in KB to approximately know the number of page it takes.

Reply:
“The Devil is in the details, but so is salvation.”
― Hyman G. Rickover

Reply:
OK I got it but....

![image](https://user-images.githubusercontent.com/18289987/208432117-90ed9206-8951-4842-8869-83f7661efa44.png)